src/backend/distributed/commands/index.c

```diff
@@ -556,16 +556,34 @@ PreprocessReindexStmt(Node *node, const char *reindexCommand,
 			state.locked_table_oid = InvalidOid;
 
 			Oid indOid = RangeVarGetRelidExtended(reindexStatement->relation,
-												  lockmode, 0,
+												  lockmode, RVR_MISSING_OK,
 												  RangeVarCallbackForReindexIndex,
 												  &state);
+			if (!OidIsValid(indOid))
```

Comment:
I think we can extract this class of changes to another PR as well: Properly check existence of objects in Preprocess functions